pkg/actuators/machine/reconciler.go

```diff
@@ -314,6 +313,7 @@ func (r *Reconciler) requeueIfInstanceBuilding(instance *models.PVMInstance) err
 	// we get a public IP populated more quickly.
 	if instance.Status != nil && *instance.Status == client.InstanceStateNameBuild {
 		klog.Infof("%s: Instance state still building, returning an error to requeue", r.machine.Name)
+		r.machineScope.setProviderStatus(instance, conditionBuild())
```

Comment:
@Karthik-K-N This instance obj which you are passing is supplied by the `launchInstance` function, will it have the correct ID of the desired machine always, since that obj is created by iterating over the list of Instances. Just want know if that list of Instances is reliable and always have the desired machine ID available in that list (assuming it wont have that delay of adding the machine to list post create and provide us with outdated list obj).

Reply:
Yes, [CreateInstance](https://github.com/Karthik-K-N/machine-api-provider-powervs/blob/0288234f0fe22a723cf0c829eddd3a6e4a25ba7c/pkg/actuators/machine/instances.go#L81-L100) call always returns the list with only one object that is the created instance. It wont return all the existing instances.

Reply:
okay so in that case do we need this [snippet](https://github.com/Karthik-K-N/machine-api-provider-powervs/blob/0288234f0fe22a723cf0c829eddd3a6e4a25ba7c/pkg/actuators/machine/instances.go#L86-L90) if it returns only one obj, can use it directly.

Reply:
I think since its pointer over slice, thats the better way to extract the instance id, Do you have any other thoughts?

Reply:
Or this is the way
```

	instances, err := client.CreateInstance(params)
	if err != nil {
		return nil, mapierrors.CreateMachine("error creating powervs instance: %v", err)
	}

	if instances == nil || len(*instances) != 1 {
		return nil, mapierrors.CreateMachine("error getting the instance ID post deployment for: %s", machine.Name)
	}

	instance, err := client.GetInstance(*(*instances)[0].PvmInstanceID)
	if err != nil {
		return nil, mapierrors.CreateMachine("error getting the instance for ID: %s", insIDs[0])
	}
	return instance, nil
```

Reply:
@Karthik-K-N you can go ahead with whichever is the best reliable approach (I would prefer using the returned obj itself)

Reply:
yeah but returned object is a list so we need to extract first element out of it.

Reply:
okay understood.